bin/demo.py

```diff
@@ -2,221 +2,215 @@
-            log_and_exit("Not authenticated. Please run `dcos auth login` and try again.")
-    except CalledProcessError:
-        log_and_exit ("Not authenticated. Please run `dcos auth login` and try again.")
+        oldstdchannel = os.dup(stdchannel.fileno())
+        dest_file = open(dest_filename, 'w')
+        os.dup2(dest_file.fileno(), stdchannel.fileno())
+        yield
+    finally:
+        if oldstdchannel is not None:
+            os.dup2(oldstdchannel, stdchannel.fileno())
+        if dest_file is not None:
+            dest_file.close()
+
+def check_and_set_token():
+    dcos_username = arguments['--dcos-username']
+    dcos_password = arguments['--dcos-password']
+    try:
+        token = shakedown.authenticate(dcos_username, dcos_password)
+        dcos.config.set_val('core.dcos_acs_token', token)
+    except:
+        log_and_exit('!! authentication failed')
 
 def config_dcos_cli(dcos_url):
-    if call (["dcos", "config", "set", "core.dcos_url", dcos_url],stdout=open(os.devnull, 'wb')) == 1:
-        log_and_exit ("Unable to configure DCOS CLI.")
-
-def make_temp_dir():
-    remove_temp_dir()
-    os.mkdir("tmp")
-
-def rename(path, jenkins_name):
-    with open(path, 'r+') as f:
-        config = json.load(f)
-        config['jenkins']['framework-name'] = jenkins_name
-        f.seek(0)
-        json.dump(config, f, indent=4)
-
-def install(dcos_url, jenkins_name, jenkins_url):
-    log ("Installing Jenkins with name {}.".format(jenkins_name))
-    shutil.copyfile("conf/jenkins.json", "tmp/jenkins.json")
-    rename("tmp/jenkins.json", jenkins_name)
-    command = "dcos package install --yes --options=tmp/jenkins.json jenkins"
-    print ("\n> " + command)
-    if call (['dcos', 'package', 'install', '--yes', '--options=tmp/jenkins.json', 'jenkins']) != 0:
-        log_and_exit ("Failed to install Jenkins.")
-    log("Jenkins has been installed! Wait for it to come up before proceeding at: {}".format(jenkins_url))
-
-def verify(jenkins_url):
-    r = requests.get(jenkins_url, headers=auth_func({}), verify=False)
-    if r.status_code != 200:
-        log ("Couldn't find a Jenkins instance running at {}.".format(jenkins_url))
+    dcos.config.set_val('core.dcos_url', dcos_url)
+    dcos.config.set_val('core.ssl_verify', 'False')
+
+def install_jenkins(jenkins_name, jenkins_url):
+    log("installing Jenkins with name '{}'".format(jenkins_name))
+    with open("conf/jenkins.json") as options_file:
+        package_config = options_file.read().replace("JENKINS_NAME", jenkins_name)
+    with open("jenkins_config.json", "w") as options_file:
+        options_file.write(package_config)
+    install_package('jenkins', None, jenkins_name, "jenkins_config.json")
+    os.remove("jenkins_config.json")
+    assert package_installed('jenkins', jenkins_name), log_and_exit('!! package failed to install')
+    log("waiting for Jenkins service to come up at '{}'".format(jenkins_url))
+    end_time = time.time() + 300
```

Comment:
Let's reduce this timeout. A minute should be sufficient for Marathon to deploy and run this application.